pkg/apis/pipeline/v1beta1/resultref.go

```diff
@@ -44,20 +44,20 @@ const (
 	// ResultResultPart Constant used to define the "results" part of a pipeline result reference
 	ResultResultPart = "results"
 	// TODO(#2462) use one regex across all substitutions
-	// variableSubstitutionFormat matches format like $result.resultname, $result.resultname[int] and $result.resultname[*]
-	variableSubstitutionFormat = `\$\([_a-zA-Z0-9.-]+(\.[_a-zA-Z0-9.-]+)*(\[([0-9]+|\*)\])?\)`
-	// exactVariableSubstitutionFormat matches strings that only contain a single reference to result or param variables, but nothing else
+	// resultVariableSubstitutionFormat matches format like $result.resultname, $result.resultname[int] and $result.resultname[*]
+	resultVariableSubstitutionFormat = `\$\([_a-zA-Z0-9.-]+(\.[_a-zA-Z0-9.-]+)*(\[([0-9]+|\*)\])?\)`
+	// isolatedResultVariableSubstitutionFormat matches strings that only contain a single reference to result or param variables, but nothing else
 	// i.e. `$(result.resultname)` is a match, but `foo $(result.resultname)` is not.
-	exactVariableSubstitutionFormat = `^\$\([_a-zA-Z0-9.-]+(\.[_a-zA-Z0-9.-]+)*(\[([0-9]+|\*)\])?\)$`
+	isolatedResultVariableSubstitutionFormat = `^\$\([_a-zA-Z0-9.-]+(\.[_a-zA-Z0-9.-]+)*(\[([0-9]+|\*)\])?\)$`
 	// arrayIndexing will match all `[int]` and `[*]` for parseExpression
 	arrayIndexing = `\[([0-9])*\*?\]`
 	// ResultNameFormat Constant used to define the the regex Result.Name should follow
 	ResultNameFormat = `^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$`
 )
 
-// VariableSubstitutionRegex is a regex to find all result matching substitutions
-var VariableSubstitutionRegex = regexp.MustCompile(variableSubstitutionFormat)
-var exactVariableSubstitutionRegex = regexp.MustCompile(exactVariableSubstitutionFormat)
+// ResultVariableSubstitutionRegex is a regex to find all result matching substitutions
+var ResultVariableSubstitutionRegex = regexp.MustCompile(resultVariableSubstitutionFormat)
```

Comment:
does this need to be exported? based on its usage in the TaskRun reconciler, it looks like it would be more useful as part of a validation helper function.